cmd/main.go

```diff
@@ -0,0 +1,42 @@
+package main
+
+import (
+	"log"
+	"net/http"
+
+	"github.com/gorilla/mux"
+
+	gwi "github.com/josedelrio85/platform-go-challenge/internal"
+)
+
+func main() {
+	log.Println("GWI Go Coding Challenge starting...")
+
+	repo, err := gwi.NewMemoryRepository()
+	if err != nil {
+		panic(err)
+	}
+	handler := gwi.Handler{
+		Repo: repo,
+	}
+
+	router := mux.NewRouter()
+
+	router.PathPrefix("/user/{userid:[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}}/fav/list").
+		Handler(handler.GetFavsFromUser()).
+		Methods(http.MethodGet)
+
+	router.PathPrefix("/user/{userid:[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}}/fav/add").
```

Comment:
I have implemented this in this way because I took into consideration that a favourite is an entity of its own that can have many forms, so the endpoint should be unique for any of the possible implementations of a favourite.

In the case of using different endpoints for each type of bookmark you would have a more atomised logic, but you would have to implement a set of CRUD endpoints for each type of bookmark, with its subsequent logic.